src/app/RecordingMetadata/ClickableLabel.tsx

```diff
@@ -0,0 +1,88 @@
+ * (b) any piece of software and/or hardware listed in the lrgrwrks.txt file if
+ * one is included with the Software (each a "Larger Work" to which the Software
+ * is contributed by such licensors),
+ *
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { getLabelDisplay } from "@app/Recordings/Filters/LabelFilter";
+import { Label } from "@patternfly/react-core";
+import React from "react";
+import { RecordingLabel } from "./RecordingLabel";
+
+
+export interface ClickableLabelCellProps {
+  label: RecordingLabel;
+  isSelected: boolean;
+  onLabelClick: (label: RecordingLabel) => void
+}
+
+export const ClickableLabel: React.FunctionComponent<ClickableLabelCellProps> = (props) => {
+  const [isHoveredOrFocused, setIsHoveredOrFocused] = React.useState(false);
+  const labelColor = React.useMemo(() => props.isSelected? "blue": "grey", [props.isSelected]);
+
+  const handleHoveredOrFocused = React.useCallback(() => setIsHoveredOrFocused(true), [setIsHoveredOrFocused]);
+  const handleNonHoveredOrFocused = React.useCallback(() => setIsHoveredOrFocused(false), [setIsHoveredOrFocused]);
+
+  const style = React.useMemo(() => {
+    if (isHoveredOrFocused) {
+      const defaultStyle = { cursor: "pointer", "--pf-c-label__content--before--BorderWidth": "2.5px"};
+      if (props.isSelected) {
+        return {...defaultStyle, "--pf-c-label__content--before--BorderColor": "#06c"}
+      }
+      return {...defaultStyle, "--pf-c-label__content--before--BorderColor": "#8a8d90"}
+    }
+    return {};
+  }, [props.isSelected, isHoveredOrFocused]);
+
+  const handleLabelClicked = React.useCallback(
+    () => props.onLabelClick(props.label), 
+    [props.label, props.onLabelClick, getLabelDisplay]
+  );
+
+  return <>
+    <Label
+      aria-label={`${props.label.key}: ${props.label.value}`}
+      style={style}
+      onMouseEnter={() => handleHoveredOrFocused()}
+      onMouseLeave={handleNonHoveredOrFocused}
```

Comment:
Is there a particular reason that this one is passed as a reference to a callback and the line above passes a lambda that calls the callback?

Reply:
Ahh no actually. I think I forgot to remove it. Fixed now.